File: data/src/app_config.py
import yaml
import os
import logging

logger = logging.getLogger(__name__)

# Дефолтные значения
_DEFAULT_CONFIG = {
	"app": {"python_stop_timeout_ms": 2000, "default_lists_path": ""},
	"terminal": {
		"theme": "Dark",
		"font_family": "Cascadia Code",
		"font_size": 12,
		"max_proc_width": 24,
		"max_ip_width": 45,
		"max_port_width": 6,
		"max_domain_width": 50,
		"color_console": True,
		"skip_local_ips": True,
		"highlight_style": "BRIGHT_WHITE",
	},
	"monitor": {
		"dns_resolve_statuses": [
			"SYN_SENT",
			# "SYN_RECV",
			# "ESTABLISHED",
			# "FIN_WAIT1",
			# "FIN_WAIT2",
			# "TIME_WAIT",
			# "CLOSE",
			# "CLOSE_WAIT",
			# "LAST_ACK",
			# "LISTEN",
			# "CLOSING",
			# "NONE"
		]
	},
	"logging": {"level": "INFO", "max_file_size": 1048576, "backup_count": 5},
}

_config = None


def load_app_config(config_dir: str = "data") -> dict:
	"""Загружает общий конфиг из data/app_config.yaml."""
	global _config
	if _config is not None:
		return _config

	path = os.path.join(config_dir, "app_config.yaml")
	if not os.path.exists(path):
		logger.warning(
			f"app_config.yaml не найден, использую значения по умолчанию: {path}"
		)
		_config = _DEFAULT_CONFIG.copy()
		return _config

	try:
		with open(path, "r", encoding="utf-8") as f:
			loaded = yaml.safe_load(f)
			if not loaded:
				_config = _DEFAULT_CONFIG.copy()
			else:
				_config = _DEFAULT_CONFIG.copy()
				_merge_dict(_config, loaded)
		logger.info(f"Загружен app_config.yaml")
	except Exception as e:
		logger.error(f"Ошибка загрузки app_config.yaml: {e}, использую дефолты")
		_config = _DEFAULT_CONFIG.copy()
	return _config


def _merge_dict(base, override):
	"""Рекурсивное слияние словарей."""
	for key, value in override.items():
		if key in base and isinstance(base[key], dict) and isinstance(value, dict):
			_merge_dict(base[key], value)
		else:
			base[key] = value


def get_app_config() -> dict:
	if _config is None:
		load_app_config()
	return _config

File: PlatformLauncher/Domain/Models/AppConfig.cs
using System.Collections.Generic;
using YamlDotNet.Serialization;

namespace PlatformLauncher.Domain.Models
{
    public class AppConfig
    {
        [YamlMember(Alias = "app")]
        public AppSettings App { get; set; }

        [YamlMember(Alias = "terminal")]
        public TerminalSettings Terminal { get; set; }

        [YamlMember(Alias = "monitor")]
        public MonitorSettings Monitor { get; set; }

        [YamlMember(Alias = "logging")]
        public LoggingSettings Logging { get; set; }

        [YamlMember(Alias = "selected_theme")]
        public string SelectedTheme { get; set; } = "fluent-light";

        [YamlMember(Alias = "lists")]
        public ListsConfig Lists { get; set; }

        [YamlMember(Alias = "cloudflare_domains")]
        public List<string> CloudflareDomains { get; set; }
    }

    public class AppSettings
    {
        [YamlMember(Alias = "app_version")]
        public string AppVersion { get; set; } = "1.0.0";

        [YamlMember(Alias = "python_stop_timeout_ms")]
        public int PythonStopTimeoutMs { get; set; } = 2000;

        [YamlMember(Alias = "default_lists_path")]
        public string DefaultListsPath { get; set; } = "";
    }

    public class TerminalSettings
    {
        [YamlMember(Alias = "theme")]
        public string Theme { get; set; } = "Dark";

        [YamlMember(Alias = "font_family")]
        public string FontFamily { get; set; } = "Cascadia Code";

        [YamlMember(Alias = "font_size")]
        public int FontSize { get; set; } = 12;

        [YamlMember(Alias = "max_proc_width")]
        public int MaxProcWidth { get; set; } = 24;

        [YamlMember(Alias = "max_ip_width")]
        public int MaxIpWidth { get; set; } = 45;

        [YamlMember(Alias = "max_port_width")]
        public int MaxPortWidth { get; set; } = 6;

        [YamlMember(Alias = "max_domain_width")]
        public int MaxDomainWidth { get; set; } = 50;

        [YamlMember(Alias = "color_console")]
        public bool ColorConsole { get; set; } = true;

        [YamlMember(Alias = "skip_local_ips")]
        public bool SkipLocalIps { get; set; } = true;

        [YamlMember(Alias = "highlight_style")]
        public string HighlightStyle { get; set; } = "BRIGHT_WHITE";
    }

    public class MonitorSettings
    {
        [YamlMember(Alias = "dns_resolve_statuses")]
        public List<string> DnsResolveStatuses { get; set; } = new List<string> { "SYN_SENT" };
    }

    public class LoggingSettings
    {
        [YamlMember(Alias = "level")]
        public string Level { get; set; } = "INFO";

        [YamlMember(Alias = "max_file_size")]
        public int MaxFileSize { get; set; } = 1048576;

        [YamlMember(Alias = "backup_count")]
        public int BackupCount { get; set; } = 5;

        [YamlMember(Alias = "debug_enabled")]
        public bool DebugEnabled { get; set; } = false;
    }
}